src/cargo/core/compiler/build_context/mod.rs

```diff
@@ -50,7 +52,31 @@ impl<'a, 'cfg> BuildContext<'a, 'cfg> {
         profiles: &'a Profiles,
         extra_compiler_args: HashMap<Unit<'a>, Vec<String>>,
     ) -> CargoResult<BuildContext<'a, 'cfg>> {
-        let rustc = config.rustc(Some(ws))?;
+        let mut rustc = config.load_global_rustc(Some(ws))?;
+
+        if build_config.clippy_override {
+            let tool = config.get_tool("clippy-driver")?;
+            let tool = paths::resolve_executable(&tool).map_err(|e| {
+                let rustup_in_path = config
+                    .get_tool("rustup")
+                    .and_then(|tool| paths::resolve_executable(&tool))
+                    .is_ok();
+                let has_rustup_env = std::env::var("RUSTUP_TOOLCHAIN").is_ok();
+                if rustup_in_path || has_rustup_env {
+                    failure::format_err!("{}: please run `rustup component add clippy`", e)
```

Comment:
FWIW I don't think that this is actually going to catch many cases, because `resolve_executable` will look at PATH, but rustup *always* installs `cargo-clippy` and `clippy-driver` as executables and they just happen to return errors if the component isn't available. I think this is a case where we probably want to defer to rustup to print messages about adding clippy